test/src/test_endian_stream.cpp

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Steinwurf ApS 2016.
+// All Rights Reserved
+//
+// Distributed under the "BSD License". See the accompanying LICENSE.rst file.
+
+#include <endian/endian_stream.hpp>
+#include <gtest/gtest.h>
+
+TEST(EndianStream, Convert)
+{
+
```

Comment:
We should probably avoid all this duplication in test_endian_stream_big_endian.cpp and test_endian_stream_little_endian.cpp. You could make more templated helper functions to minimize duplication. I would put all tests in test_endian_stream.cpp to follow our naming guidelines.